### PR TITLE
Use Infra terms to simplify text.

### PR DESCRIPTION
**Pull request by inexorabletash**
* Define array keys as "byte sequence" and compare with "byte less than"
* Compare strings using "code unit less than"
* Make a few lists/sets the Infra types, and use items, contains, append, and so on.
* Simplify key comparison steps for non-equal types.
* Linkify "Assert:"

No normative changes.

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/IndexedDB/pull/275.html" title="Last updated on Jun 12, 2019, 9:42 PM UTC (8499394)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/IndexedDB/275/ce86e18...8499394.html" title="Last updated on Jun 12, 2019, 9:42 PM UTC (8499394)">Diff</a>

### CODE REVIEW

**Comment by inexorabletash** [comment]
The behavior is exactly what was present previously. Sorting is identical to string sorting in JavaScript.

**Comment by aphillips** [comment]
Okay, cool. I mean, it's not ideal, but it makes sense.

**Comment by inexorabletash** [comment]
For the record, #38 tracks locale-aware sorting.

**Comment by aphillips** [comment]
@inexorabletash Yes. In this case, though, I was concerned that code point sorting is better than code unit sorting. In any case, your explanation makes sense.

**Comment by inexorabletash** [comment]
@aliams - can you take a look at this one too?

(It'll conflict with the other one, so whichever comes second will need a rebase)

File: index.bs [hunk 1]
```diff
@@ -688,50 +677,32 @@ To <dfn>compare two keys</dfn> |a| and |b|, run these steps:
...
       <dt>*array*</dt>
       <dd>
-        1. Let |length| be the lesser of |va|'s length and |vb|'s length.
+        1. Let |length| be the lesser of |va|'s [=list/size=] and |vb|'s [=list/size=].
         1. Let |i| be 0.
         1. While |i| is less than |length|, then:
 
-            1. Let |u| be the [=/key=] in |va| at index |i|.
-            1. Let |v| be the [=/key=] in |vb| at index |i|.
+            1. Let |u| be the [=/key=] at |va|[|i|].
```
Comment (aliams) [inline index.bs]: At or in?

Comment (inexorabletash) [reply index.bs]: I was thinking the temporary _u_ and _v_ variables could be removed entirely, which makes this moot. I'll do that.

**Pull request merged**